server/webhook_parser.go

```diff
@@ -405,16 +400,74 @@ func preProcessText(jiraMarkdownString string) string {
 		return "> " + quotedText
 	})
 
-	processedString = noFormatRegex.ReplaceAllStringFunc(processedString, func(noFormatBlock string) string {
-		content := noFormatBlock[strings.Index(noFormatBlock, "}")+1 : strings.LastIndex(noFormatBlock, "{noformat}")]
-		return fmt.Sprintf("`%s`", content)
-	})
-
 	processedString = doubleCurlyRegex.ReplaceAllStringFunc(processedString, func(match string) string {
 		content := match[2 : len(match)-2]
 		return fmt.Sprintf("`%s`", content)
 	})
 
+	// handles sinlge and multi line language specific code blocks
+	processedString = langSpecificCodeBlockRegex.ReplaceAllStringFunc(processedString, func(langSpecificBlock string) string {
+		startIndex := strings.Index(langSpecificBlock, "{code:")
+		endIndex := strings.LastIndex(langSpecificBlock, "{code}")
+		if startIndex == -1 || endIndex == -1 || startIndex == endIndex {
+			return langSpecificBlock
+		}
+
+		langEndIndex := strings.Index(langSpecificBlock[startIndex:], "}")
+		if langEndIndex == -1 {
+			return langSpecificBlock
+		}
+
+		contentStartIndex := startIndex + langEndIndex + 1
+		content := langSpecificBlock[contentStartIndex:endIndex]
+
+		lines := strings.Split(content, "\n")
+		for i := range lines {
+			if lines[i] != "" {
+				lines[i] = "`" + lines[i] + "`"
+			}
+		}
+
+		if len(lines) == 1 {
+			return "`" + lines[0] + "`"
+		}
+
+		return "\n" + strings.Join(lines, "\n") + "\n"
+	})
+
+	// handles sinlge and multi line non-language specific code blocks
+	processedString = noFormatRegex.ReplaceAllStringFunc(processedString, func(noFormatBlock string) string {
+		startIndex := strings.Index(noFormatBlock, "{noformat}")
+		endIndex := strings.LastIndex(noFormatBlock, "{noformat}")
+		if startIndex == -1 || endIndex == -1 || startIndex == endIndex {
+			return noFormatBlock
+		}
+
+		content := noFormatBlock[startIndex+len("{noformat}") : endIndex]
+
+		lines := strings.Split(content, "\n")
+
+		if len(lines) > 0 {
+			lines[0] = "`" + lines[0] + "`\n"
+		}
+
+		for i := 1; i < len(lines)-1; i++ {
+			if lines[i] != "" {
+				lines[i] = "`" + lines[i] + "`"
+			}
+		}
+
+		if len(lines) > 1 {
+			lines[len(lines)-1] = "`" + strings.TrimSuffix(lines[len(lines)-1], "{noformat}") + "`"
+		}
+
+		if len(lines) == 1 {
+			return "`" + lines[0] + "`"
+		}
```

Comment:
This should've been covered by lines 450-452.

Reply:
Required condition to check to make sure no `\n` is added to single line of block code.
can move this code block below line 448 if we want

Reply:
Okay, move this to below line 448 and make the same change in the first comment if possible.

Reply:
Possible, doing it now